crates/query-engine/translation/src/translation/mutation/delete.rs

```diff
@@ -0,0 +1,90 @@
+use query_engine_metadata::metadata;
+use query_engine_sql::sql::ast;
+
+// this can get us `DELETE FROM <table> WHERE column = <column_name_arg>`
+// TODO: think about how RETURNING should work - ideally we'd not bake the columns in,
```

Comment:
imo just put `RETURNING *`. The query part will deal with selecting the relevant columns.